### PR TITLE
feat: add some better value selection when providing multiple keys

### PR DESCRIPTION
**Pull request by domharrington**
## 🧰 Changes

- Return with a value from the top level if it's not set in the keys[]
- Return with a default if it's not set in the keys[]
- Do not render a dropdown if the value hasn't come from the keys[]
- Do not render a dropdown if there's only one value in the keys[]

## 🧬 QA & Testing

Do the tests pass? ✅ 
I've tested this npm linked into ReadMe but I wanna do a little more full QA to ensure all of these different edge cases are working as expected over there.

### CODE REVIEW

**Review by gratcliff — APPROVED** [review]
That's quite the change! Code looks solid to me.

File: index.jsx [hunk 1]
```diff
@@ -27,18 +27,42 @@ class Variable extends React.Component {
...
+
+  // Determining whether to show a dropdown or not onClick
+  // based upon whether the currently displayed value has come
+  // from one of the nested user.keys[] or not.
+  // Additionally do not show the dropdown if there is only a
+  // single key, since there's nothing to change it to.
+  shouldShowVarDropdown(selected) {
+    const { user, variable } = this.props;
+
+    return !!this.getSelectedValue(selected)[variable] && Array.isArray(user.keys) && user.keys.length > 1;
```
Comment (kellyjosephprice) [inline index.jsx:47]: Should it only count keys with values? Is that possible?
```suggestion
    return !!this.getSelectedValue(selected)[variable] && Array.isArray(user.keys) && user.keys.filter(key => key.name === variable).length > 1;
```

Comment (domharrington) [reply index.jsx:47]: We're already checking if this particular variable has a value assigned in the first part of this statement:

```
!!this.getSelectedValue(selected)[variable]
```

Is that sufficient for the use case you're referring to? If there isn't a `user.keys[variable]` then it'll already have exited early from that statement because we already know not to show the var dropdown.

Comment (kellyjosephprice) [reply index.jsx:47]: Gotcha gotcha, misread that.

**Review by erunion — APPROVED** [review]

**Comment by erunion** [comment]
lgtm, @domharrington want me to merge and tag a release for you?

**Comment by domharrington** [comment]
@erunion yes please 🙏
